File: R Code/HighridgeConstructionCompanyPayment Slips.R
# Define a function that takes in the number of employees
createWorkers <- function(numWorkers) {
  workers <- list() # Empty list to hold employee data
  
  # loop through the number of employees to populate list
  for (x in 1:numWorkers){
    # create dynamic ID's
    workerID <- sprintf("employee_%04d", x) 
    
    # create random salaries
    workerSalary <- sample(5000:40000, 1)
    
    # randomly create gender
    workerGender <- sample(c('Male', 'Female', 'Other'), 1)
    
    # try to choose employee level
    
    employeeLevel <- tryCatch({
      if(workerSalary > 10000 && workerSalary < 20000){
        "A1"
      } else if (workerSalary > 7500 && workerSalary < 30000 && workerGender == 'Female'){
        "A5-F"
      } else {
        "Unspecified"
      }
    }, error = function(e) {
      "Unknown"   #handle exception
    }
    )
    
    # Creating list for each employee
    worker <- list(id = workerID, salary = workerSalary, gender = workerGender, level = employeeLevel)
    
    # adding workers to the workers list 
    workers[[x]] <- worker
    
  }
  
  return(workers)
  
}

# Define variable for number of employees
numEmployees <- 500

# call function on number of employees
employeeList <- createWorkers(numEmployees)

# create table for better presentation of data
df <- do.call(rbind, lapply(employeeList, as.data.frame))
print(df)

File: Python Code/Highridge Construction Company Payment Slips.py
#import random module
import random
# define a function that takes a number representing the number of employees
def create_workers(num_workers):
    workers = []
    # loop through the number of employees and create a dictionary for each
    for x in range(num_workers):
        worker_id = f"employee_{x+1:04d}"
        worker_salary = random.randint(5000,40000)
        worker_gender = random.choice(['Male','Female','Other'])
        # try to create condition for choosing employee level
        try:
            if 10000 < worker_salary < 20000:
                employee_level = "A1"
            elif 7500 < worker_salary < 30000 and worker_gender == 'Female':
                employee_level = "A5-F"  
            else:
                employee_level = "Unspecified"
        # handle exceptions 
        except Exception:
            employee_level = "Unknown"
        # create a dictionary for each employee
        worker = {"id": worker_id, "salary": worker_salary, "gender": worker_gender, 'level': employee_level}
        # add worker to employee list
        workers.append(worker)
    return workers

# define a variable to hold the number of employees
num_workers = 500

# call the function to create the employees
create_workers(num_workers)